File: app/src/main/java/org/amfoss/paneeer/editorCamera/util/PreferenceUtil.java
package org.amfoss.paneeer.editorCamera.util;

import android.content.Context;
import android.content.SharedPreferences;

public class PreferenceUtil {

    public static void putLongValue(Context context, String prefName, String key, long value) {
        SharedPreferences pref = context.getSharedPreferences(prefName, 0);
        SharedPreferences.Editor editor = pref.edit();
        editor.putLong(key, value);
        editor.apply();
    }

    public static long getLongValue(Context context, String prefName, String key) {
        SharedPreferences pref = context.getSharedPreferences(prefName, 0);
        try {
            return pref.getLong(key, 0);
        } catch (Exception e) {
            return 0;
        }
    }

    public static void putStringValue(Context context, String prefName, String key, String value) {
        SharedPreferences pref = context.getSharedPreferences(prefName, 0);
        SharedPreferences.Editor editor = pref.edit();
        editor.putString(key, value);
        editor.apply();
    }

    public static String getStringValue(Context context, String prefName, String key) {
        try {
            SharedPreferences pref = context.getSharedPreferences(prefName, 0);
            return pref.getString(key, null);
        } catch (Exception e) {
            return null;
        }
    }
}

File: app/src/main/java/org/amfoss/paneeer/gallery/util/PreferenceUtil.kt
package org.amfoss.paneeer.gallery.util

import android.content.Context
import android.content.SharedPreferences
import android.preference.PreferenceManager

class PreferenceUtil private constructor(mContext: Context) {
    private val SP: SharedPreferences
    val editor: SharedPreferences.Editor
        get() = SP.edit()

    fun putString(key: String?, value: String?) {
        editor.putString(key, value).commit()
    }

    fun getString(key: String?, defValue: String?): String? {
        return SP.getString(key, defValue)
    }

    fun putInt(key: String?, value: Int) {
        editor.putInt(key, value).commit()
    }

    fun getInt(key: String?, defValue: Int): Int {
        return SP.getInt(key, defValue)
    }

    fun putBoolean(key: String?, value: Boolean) {
        editor.putBoolean(key, value).commit()
    }

    fun getBoolean(key: String?, defValue: Boolean): Boolean {
        return SP.getBoolean(key, defValue)
    }

    fun remove(key: String?) {
        editor.remove(key).commit()
    }

    fun clearPreferences() {
        editor.clear().commit()
    }

    companion object {
        private var instance: PreferenceUtil? = null
        @JvmStatic
        fun getInstance(context: Context): PreferenceUtil? {
            if (instance == null) {
                synchronized(
                    PreferenceUtil::class.java
                ) {
                    if (instance == null) instance =
                        PreferenceUtil(context)
                }
            }
            return instance
        }
    }

    init {
        SP = PreferenceManager.getDefaultSharedPreferences(mContext)
    }
}